ingest/ingestHelpers.py

```diff
@@ -139,121 +125,44 @@ def get_distributions(ds):
-####################################################
-##
-####################################################
+#############################################
+#    Implementation of main(...):
+#
+def main(get_objects_query_location, describe_object_query_location,
+         elasticsearch_index, elasticsearch_type, variable_name_sparql, XIngest):
 
-def load_file(filepath):
-    with open(filepath) as _file:
-        return _file.read().replace('\n', " ")
-
-
-
-def select(endpoint, query):
-    sparql = SPARQLWrapper(endpoint)
-    sparql.setQuery(query)
-    sparql.setReturnFormat(JSON)
-    results = sparql.query().convert()
-    return results["results"]["bindings"]
-
-
-# get_datasets -> get_objects
-def get_objects(endpoint, get_objects_query, object_type):
-    r = select(endpoint, get_objects_query)
-    # return [rs["dataset"]["value"] for rs in r]
-    return [rs[object_type]["value"] for rs in r]
-
-
-def describe(endpoint, query):
-    sparql = SPARQLWrapper(endpoint)
-    sparql.setQuery(query)
-    try:
-        return sparql.query().convert()
-    except RuntimeWarning:
-        pass
-
-
-# process_dataset: used by generate
-def process_dataset(dataset, endpoint, create_object_doc_function, object_index, object_type):
-    # ds = create_dataset_doc(dataset=dataset, endpoint=endpoint) ==>
-    ds = create_object_doc_function(dataset=dataset, endpoint=endpoint)
-    if "dcoId" in ds and ds["dcoId"] is not None:
-        return [json.dumps(get_metadata(object_index, object_type, (ds["dcoId"]))), json.dumps(ds)]
-    else:
-        return []
-
-
-def generate(threads, sparql, get_objects_query, process_object_function, create_object_doc_function, object_index, object_type):
-    pool = multiprocessing.Pool(threads)
-    # params = [(dataset, sparql) for dataset in get_datasets(endpoint=sparql)] ==>
-    params = [(object, sparql, create_object_doc_function, object_index, object_type) for object in get_objects(endpoint=sparql, get_objects_query=get_objects_query, object_type=object_type)]
-    return list(itertools.chain.from_iterable(pool.starmap(process_object_function, params)))
-
-def Main(get_objects_query, create_object_doc_function, object_index, object_type):
     parser = argparse.ArgumentParser()
-    parser.add_argument('--threads', default=1, help='number of threads to use (default = 8)')
+    parser.add_argument('--threads', default=4, help='number of threads to use (default = 4)')
     parser.add_argument('--es', default="http://localhost:9200", help="elasticsearch service URL")
-    # parser.add_argument('--es', default="https://dcotest.tw.rpi.edu/search", help="elasticsearch service URL")
     parser.add_argument('--publish', default=False, action="store_true", help="publish to elasticsearch?")
     parser.add_argument('--rebuild', default=False, action="store_true", help="rebuild elasticsearch index?")
     parser.add_argument('--mapping', default="mappings/dataset.json", help="dataset elasticsearch mapping document")
```

Comment:
Well, we can use the class attributes. But is it having the mapping passed in a more "maneuverable" approach?